src/coreclr/vm/jitinterface.cpp

```diff
@@ -1697,16 +1742,50 @@ void CEEInfo::getThreadLocalStaticBlocksInfo (CORINFO_THREAD_STATIC_BLOCKS_INFO*
 
     JIT_TO_EE_TRANSITION_LEAF();
 
-    pInfo->tlsIndex.addr = (UINT8*)0;
+    size_t threadStaticBaseOffset = 0;
 
-    pInfo->offsetOfThreadLocalStoragePointer = 0;
-    pInfo->offsetOfThreadStaticBlocks = 0;
-    pInfo->offsetOfMaxThreadStaticBlocks = 0;
-    pInfo->offsetOfGCDataPointer = 0;
+#if defined(TARGET_WINDOWS)
+    pInfo->tlsIndex.addr = (void*)static_cast<uintptr_t>(_tls_index);
+    pInfo->tlsIndex.accessType = IAT_VALUE;
 
-    JIT_TO_EE_TRANSITION_LEAF();
+    pInfo->offsetOfThreadLocalStoragePointer = offsetof(_TEB, ThreadLocalStoragePointer);
+    threadStaticBaseOffset = ThreadLocalOffset(&t_ThreadStatics);
+
+#elif defined(TARGET_OSX)
+
+    pInfo->threadVarsSection = getThreadVarsSectionAddress();
+
+#elif defined(TARGET_AMD64)
+
+    // For Linux/x64, get the address of tls_get_addr system method and the base address
+    // of struct that we will pass to it.
+    pInfo->tlsGetAddrFtnPtr = reinterpret_cast<void*>(&__tls_get_addr);
+    pInfo->tlsIndexObject = getTlsIndexObjectAddress();
+
+#elif defined(TARGET_ARM64)
+
+    // For Linux/arm64, just get the offset of thread static variable, and during execution,
+    // this offset, taken from trpid_elp0 system register gives back the thread variable address.
+    threadStaticBaseOffset = reinterpret_cast<size_t>(GetThreadStaticsVariableOffset());
```

Comment:
```suggestion
    threadStaticBaseOffset = GetThreadStaticsVariableOffset();
```
and change it to return the right type directly.